pkg/fqdn/dnsproxy/proxy.go

```diff
@@ -119,9 +120,26 @@ type LookupEndpointIDByIPFunc func(ip net.IP) (endpointID string, err error)
 
 // NotifyOnDNSMsgFunc handles propagating DNS response data
 // See DNSProxy.LookupEndpointIDByIP for usage.
-type NotifyOnDNSMsgFunc func(lookupTime time.Time, srcAddr, dstAddr string, msg *dns.Msg, protocol string, allowed bool, proxyErr error) error
+type NotifyOnDNSMsgFunc func(lookupTime time.Time, srcAddr, dstAddr string, msg *dns.Msg, protocol string, allowed bool, stat ProxyRequestContext) error
+
+// ProxyRequestContext proxy dns request context struct to send in the callback
+type ProxyRequestContext struct {
+	TotalTime    spanstat.SpanStat
```

Comment:
no, just because the TotalTime is impossible to report back because the callback does not send anything when we send the information to the dns client.